src/test/java/com/github/bibsysdev/nva/cristin/person/OneCristinPersonHandlerTest.java

```diff
@@ -0,0 +1,156 @@
+        cristinApiClientStub = Mockito.spy(cristinApiClientStub);
+        doReturn(getReader(CRISTIN_ONE_PERSON_MISSING_FIELDS_JSON_FILE)).when(cristinApiClientStub).getResponse(any());
+        handler = new OneCristinPersonHandler(cristinApiClientStub, environment);
+        GatewayResponse<NvaPerson> response = sendQueryWithId(DEFAULT_ID);
+        var expected = getReader(NVA_ONE_PERSON_MISSING_FIELDS_JSON_FILE);
+
+        assertEquals(objectMapper.readTree(expected), objectMapper.readTree(response.getBody()));
+    }
+
+    @Test
+    void handlerReturnsNvaPersonFromTransformedCristinPersonWhenIdIsFound() throws Exception {
+        GatewayResponse<NvaPerson> response = sendQueryWithId(DEFAULT_ID);
+        assertEquals(
+            objectMapper.readTree(getReader(NVA_ONE_PERSON_JSON_FILE)),
+            objectMapper.readTree(response.getBody())
+        );
+    }
+
+    @Test
+    void handlerReturnsBadGatewayExceptionWhenFetchFromBackendFails() throws Exception {
+        cristinApiClientStub = Mockito.spy(cristinApiClientStub);
+        doThrow(new IOException()).when(cristinApiClientStub).getPerson(any(), any());
+        handler = new OneCristinPersonHandler(cristinApiClientStub, environment);
+        GatewayResponse<NvaPerson> response = sendQueryWithId(DEFAULT_ID);
+
+        assertEquals(HttpURLConnection.HTTP_BAD_GATEWAY, response.getStatusCode());
+        assertEquals(APPLICATION_PROBLEM_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+
+        doThrow(new FileNotFoundException()).when(cristinApiClientStub).getPerson(any(), any());
+        handler = new OneCristinPersonHandler(cristinApiClientStub, environment);
+        response = sendQueryWithId(DEFAULT_ID);
+
+        assertEquals(HttpURLConnection.HTTP_BAD_GATEWAY, response.getStatusCode());
+        assertEquals(APPLICATION_PROBLEM_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+
+        doThrow(new BadGatewayException("")).when(cristinApiClientStub).queryOneCristinPerson(any(), any());
+        handler = new OneCristinPersonHandler(cristinApiClientStub, environment);
+        response = sendQueryWithId(DEFAULT_ID);
+
+        assertEquals(HttpURLConnection.HTTP_BAD_GATEWAY, response.getStatusCode());
+        assertEquals(APPLICATION_PROBLEM_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+    }
+
+    @Test
+    void callingHasValidContentOnCristinPersonOnlyReturnsTrueWhenAllRequiredDataArePresent() {
+        CristinPerson cristinPerson = new CristinPerson();
+        assertFalse(cristinPerson.hasRequiredFields());
+        cristinPerson.setCristinPersonId("1234");
+        cristinPerson.setFirstName("Mattias");
+        cristinPerson.setSurname("Testesen");
+        assertTrue(cristinPerson.hasRequiredFields());
+    }
+
+    private GatewayResponse<NvaPerson> sendQueryWithId(String id) throws IOException {
+        InputStream input = requestWithLanguageAndId(
```

Comment:
inline method. it is clearer if you call the requestBuilder here directly

Reply:
done